src/main/java/crawlercommons/sitemaps/extension/LinkAttributes.java

```diff
@@ -96,4 +98,20 @@ public boolean equals(Object other) {
                         && Objects.equals(params, that.params);
     }
 
+    @Override
+    public Map<String, String[]> asMap() {
+        Map<String, String[]> map = new HashMap<>();
+
+        if (href != null) {
+            map.put("href", new String[]{ href.toString() });
+        }
+
+        if (params != null) {
+
+            for (Entry<String, String> entry : params.entrySet()) {
+                map.put(entry.getKey(), new String[] { entry.getValue() });
```

Comment:
Are we guaranteed that non of these entries has a key = `href`? I think it would be safer and clearer to make the map key something like `params.xxx`

Reply:
Good call out, it is not a guarantee.  I'll prefix these parameters to avoid conflicts and potential overwrites.